iceoryx_examples/complexdata/iox_subscriber_complexdata.cpp

```diff
@@ -0,0 +1,119 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/subscriber.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+
+bool killswitch = false;
+constexpr char APP_NAME[] = "iox-cpp-subscriber-complexdata";
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // register sigHandler
+    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    // initialize subscriber
+    iox::popo::Subscriber<ComplexDataType> subscriber({"Radar", "FrontLeft", "Object"});
+
+    // run until interrupted by Ctrl-C
+    while (!killswitch)
+    {
+        subscriber.take()
+            .and_then([](auto& sample) {
+                std::stringstream s;
+                s << APP_NAME << " got values:";
+
+                s << std::endl << "stringForwardList: ";
+                for (auto i = sample->stringForwardList.begin(); i != sample->stringForwardList.end(); ++i)
+                {
+                    s << *i << " ";
+                }
+
+                s << std::endl << "integerList: ";
+                for (auto i = sample->integerList.begin(); i != sample->integerList.end(); ++i)
+                {
+                    s << *i << " ";
+                }
+
+                s << std::endl << "optionalList: ";
+                for (auto i = sample->optionalList.begin(); i != sample->optionalList.end(); ++i)
+                {
+                    (i->has_value()) ? s << i->value() << " " : s << "optional is empty, ";
+                }
+
+                s << std::endl << "floatStack: ";
+                auto stackCopy = sample->floatStack;
+                for (uint64_t i = stackCopy.capacity(); i > 0U; i--)
+                {
+                    auto result = stackCopy.pop();
+                    (result.has_value()) ? s << result.value() << " " : s << "stack is empty";
+                }
+
+                s << std::endl << "someString: ";
+                s << sample->someString;
+
+                s << std::endl << "doubleVector: ";
+                for (uint64_t i = 0U; i < sample->doubleVector.size(); ++i)
+                {
+                    s << sample->doubleVector[i] << " ";
+                }
```

Comment:
it would be nice to have this as coma separated output ... without a superfluous comma at the end of course ;)

Reply:
Well, I didn't want to add more extra code only used for the output which is not necessary for understanding how to send complex data. But no hard feelings about that. I can add it.

Reply:
Oh, maybe I got you wrong. Can you explain in more detail what you mean with `coma separated`? ;)

Reply:
no, you got it right. I meant a comma separated list, but it seems you already did it in an elegant way ... I guess I was in a coma when I wrote that :)